proofs/pseudocode-defns/pseudocode_defns.tex

```diff
@@ -0,0 +1,797 @@
+\end{definition}
+\begin{definition}[\texttt{DistanceConstant}]
+    A type \texttt{TO} has trait \texttt{DistanceConstant(TI)} if:
+    \begin{itemize}
+        \item \texttt{TO} has trait \texttt{Mul(Output=TO)} (multiplication can be done with type \texttt{TO})
+        \item \texttt{TO} has trait \texttt{Div(Output=TO)} (some form of inverse mapping can be done with type \texttt{TO})
+        \item \texttt{TO} has trait \texttt{PartialOrd} (\texttt{TO} has a partial ordering)
+        \item \texttt{TO} has trait \texttt{InfCast(TI)}
+    \end{itemize}
+    
+    \inOpenDPRust{DistanceConstant}{https://github.com/opendp/opendp/blob/main/rust/opendp/src/traits.rs}
+\end{definition}
+\begin{definition}[\texttt{Domain}]
+\label{defn:traits-domain}
+A type \texttt{T} has trait \texttt{Domain} $\iffText$ it can represent a set of values that make up a domain. The \texttt{Domain} implementation prescribes a type for members of the domain, as well as a method to check if any instance of that type is a member of that domain.
+\end{definition}
+\begin{definition}[\texttt{ExactIntCast}]
+    A type \texttt{TO} has trait \texttt{ExactIntCast(TI)} $\iffText$ Every value of type \texttt{TI} can be \texttt{exact\_int\_cast}ed exactly to a value of type \texttt{TO}, as long as the original value of type \texttt{TI} is no smaller than \texttt{get\_min\_consecutive\_int(TO)} and no larger than \texttt{get\_max\_consecutive\_int(TO)}.
+    
+    A cast error is returned when the value being \texttt{exact\_int\_cast}ed is greater than \texttt{get\_max\_consecutive\_int(TO)} or less than \texttt{get\_min\_consecutive\_int(TO)}.
+\end{definition}
+\begin{definition}[\texttt{Float}]
+    Generic trait for floating point numbers. A type \texttt{T} with trait \texttt{Float} automatically implements trait \texttt{Div<Self, Output = Self>}.
+\end{definition}
+\begin{definition}[\texttt{ImputableDomain}] 
+Any domain for which the \texttt{ImputableDomain} trait is implemented for it, has 
+\begin{itemize}
+    \item an associated type \texttt{Imputed}. \texttt{VectorDomain::Imputed} is the data type after imputation. 
+    \item an imputation function \texttt{impute\_constant} that replaces a null value with a constant or passes a non-null value through. The pseudocode is \\
+    \texttt{def impute\_constant(x: DA) -> DA::Imputed: \\ return constant if x.is\_null else x}
+    \item  a function \texttt{is\_null} to check if a value is \texttt{null}
+    \item a function \texttt{new} to construct an instance of the domain
+\end{itemize}
+\end{definition}
+\begin{definition}[\texttt{InfCast}]
+A type \texttt{TO} has trait \texttt{InfCast(TI)} $\iffText$ for every value \texttt{val} of type \texttt{TI}, \texttt{inf\_cast(val:TI,TO)} is defined (note in definition \ref{defn:inf-cast} for \texttt{inf\_cast} that it is acceptable for \texttt{inf\_cast(val:TI,TO)} to return an error).
```

Comment:
this phrasing is not clear.  is returning an error consistent with being "defined" or not?